File: cf_sdk/src/main/java/com/example/cf_sdk/logic/changebanksdk/source/GoogleDatasource.java
package com.example.cf_sdk.logic.changebanksdk.source;

import com.example.cf_sdk.logic.changebanksdk.model.google.PlaceAutocompleteSerializer;
import com.example.cf_sdk.logic.changebanksdk.model.google.PlaceDetailSerializer;

import io.reactivex.Single;

/**
 *
 * Datasource for Google Places API.
 */

public interface GoogleDatasource {
    Single<PlaceDetailSerializer> getPlaceDetail(String placeId );
    Single<PlaceAutocompleteSerializer> getPlacePredictions(String input );

}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/source/datasource/GoogleDatasource.kt
package com.example.cf_sdk.changebankapi.source.datasource

import com.example.cf_sdk.changebankapi.model.google.PlaceAutocompleteSerializer
import com.example.cf_sdk.changebankapi.model.google.PlaceDetailSerializer
import io.reactivex.Single

/**
 *
 * Datasource for Google Places API.
 */
interface GoogleDatasource {
    fun getPlaceDetail(placeId: String?): Single<PlaceDetailSerializer?>?
    fun getPlacePredictions(input: String?): Single<PlaceAutocompleteSerializer?>?
}